pkg/gds/emails/emails_test.go

```diff
@@ -90,18 +90,19 @@ func TestSendEmails(t *testing.T) {
 	// will be read, making it simpler to run tests and set environment variables.
 	godotenv.Load()
 
-	if os.Getenv("GDS_TEST_SENDING_EMAILS") == "" {
-		t.Skip("skip generate and send emails test")
-	}
-
 	// This test uses the environment to send rendered emails with context specific
 	// emails - this is to test the rendering of the emails with data only; it does not
 	// go through any of the server workflow for generating tokens, etc.
 	var conf config.EmailConfig
 	err := envconfig.Process("gds", &conf)
 	require.NoError(t, err)
 
-	// This test sends emails from the serviceEmail using SendGrid to the adminsEmail
+	if !conf.EmailTesting {
+		t.Skip("skip generate and send emails test")
+	}
```

Comment:
The way this test worked before was to actually send the emails and we used it both to verify that the `EmailManager` was working correctly and that the emails looked the way they should after being rendered from the templates. 

With the mock in place I don't think there is any reason to trigger an email send (though I guess we could keep the original test function in place called `TestSendLiveEmails` and only use it if something is wrong with email sending). Because we're not sending emails in the tests anymore, I propose that we take more control of the configuration and manually construct an `EmailConfig` without `godotenv.Load()` or a skip flag. E.g. 

```
func TestSendEmails(t *testing.T) {
    conf := config.EmailConfig{
        AdminEmail: "admin@example.com",
        ...
        Testing: true,
    }

    email, err := emails.New(conf)
    require.NoError(t, err)
}
```

So that removes quite a few lines of code! 

Also, I don't know how you feel about the 1 giant test to do all the testing for all the different emails. If it's annoying you, you could consider using a [test suite](https://pkg.go.dev/github.com/stretchr/testify/suite) here - the mock email client makes this a bit easer:

```go
import (
    "testing"
    "github.com/stretchr/testify/suite"
)

type EmailTestSuite struct {
    suite.Suite
    conf config.EmailConfig
}

func (suite *EmailTestSuite) SetupSuite() {
    suite.conf = config.EmailConfig{
        AdminEmail: "admin@example.com",
        …
        Testing: true,
    }
}

func (suite *EmailTestSuite) AfterTest(suiteName, testName string) {
    emails.PurgeMockEmails()
}

func (suite *EmailTestSuite) TestVerifyContactEmail() {
    require := suite.Require()
    email, err := emails.New(suite.conf)
    require.NoError(err)
    ...
}

func TestEmailSuite(t *testing.T) {
    suite.Run(t, &EmailTestSuite{})
}
```

This is totally optional though!